include/pstl_offload/internal/usm_memory_replacement_common.h

```diff
@@ -43,12 +43,43 @@ struct __block_header
 
 static_assert(__is_power_of_two(sizeof(__block_header)));
 
+using __realloc_func_type = void* (*)(void*, std::size_t);
+
 #if __linux__
 
+inline std::size_t
+__get_page_size()
+{
+    return sysconf(_SC_PAGESIZE);
+}
+
+inline void*
+__original_realloc(void* __user_ptr, std::size_t __new_size)
+{
+    static __realloc_func_type __orig_realloc = __realloc_func_type(dlsym(RTLD_NEXT, "realloc"));
+    return __orig_realloc(__user_ptr, __new_size);
+}
+
+#elif _WIN64
+
+std::size_t
```

Comment:
Why do we need these forward declarations? Are we moving them to the library?

Reply:
Specifically, `__get_page_size()` implementation under Windows requires `windows.h` inclusion. One is huge, and TBB users complain about having it in interface header (while this is only ETS, not every TBB header). In case of PSTL offload, `usm_memory_replacement_common.h` is included in virtually every header, so it's better to keep it as light as we can.

For `__original_realloc()`/`__original_malloc()`/`__original_aligned_alloc()`, I know no other way to call non-replaced functions.

For `__aligned_realloc_real_pointer()`, I should think more.

Reply:
Thanks, `__aligned_realloc_real_pointer()` moved to the header.